crates/red_knot/tests/file_watching.rs

```diff
@@ -42,19 +43,60 @@ impl TestCase {
     }
 
     fn stop_watch(&mut self) -> Vec<watch::ChangeEvent> {
-        if let Some(watcher) = self.watcher.take() {
-            // Give the watcher some time to catch up.
-            std::thread::sleep(Duration::from_millis(10));
-            watcher.flush();
-            watcher.stop();
+        self.try_stop_watch(Duration::from_secs(10))
+            .expect("Expected watch changes but observed none.")
+    }
+
+    fn try_stop_watch(&mut self, timeout: Duration) -> Option<Vec<watch::ChangeEvent>> {
```

Comment:
I hope this will help to improve the stability of the tests. We can now use a high timeout for functions that expect at least a single event and a smaller timeout for functions that don't.